lib/uring/uring_stubs.c

```diff
@@ -146,6 +148,37 @@ ocaml_uring_sq_ready(value v_uring) {
   return (Int_val(io_uring_sq_ready(ring)));
 }
 
+void /* noalloc */
+ocaml_uring_set_timespec(value v_sketch_ptr, value v_timeout)
+{
+  struct __kernel_timespec *t = Sketch_ptr_val(v_sketch_ptr);
+  t->tv_sec = 0;
+  t->tv_nsec = Int64_val(v_timeout);
+}
+
+value /* noalloc */
+ocaml_uring_submit_timeout(value v_uring, value v_id, value v_sketch_ptr, value v_clock, value v_rel)
+{
+  struct __kernel_timespec *t = Sketch_ptr_val(v_sketch_ptr);
+  struct io_uring* ring = Ring_val(v_uring);
+  struct io_uring_sqe* sqe;
+  int flags;
+
+  if (v_clock == caml_hash_variant("Boottime"))
```

Comment:
It looks like `caml_hash_variant` does some calculation. I'm not sure the C compiler will evaluate it at compile time. Maybe this should be calculated once as a constant?

Reply:
Is this not working as expected? My printf debuggging on this conversion seems to suggest that the values are correct, i.e. `Boottime -> IORING_TIMEOUT_BOOTTIME`, `Absolute -> IORING_TIMEOUT_ABS`. 

Another approach I tried earlier was to define these values statically using dune-configurator. However that didn't work as there is an existing dune issue where the dune-configurator can't include the vendored "liburing.h" header file. (https://ocamllabs.slack.com/archives/C16082YMP/p1658838921704329).

Reply:
It works. i just thought it might be quicker to life this out of the function, and have something like:
```suggestion
  if (v_clock == variant_boottime))
```
But it might be fast enough that it doesn't matter, or it might be that the compiler evaluates it at compile time anyway.

Reply:
There is now a regular variant `type clock = Boottime | Realtime` defined for the clocks. This allows us to use `#define Val_boottime Val_int(0)` which should help with performance.